functions/src/modules/email.functions.ts

```diff
@@ -91,10 +166,13 @@ export const sendNotificationEmail = functions.database.ref('/email/notification
     let id;
     if (Object.keys(emailTemplates).indexOf(templateName) > -1)
         id = emailTemplates[templateName].id;
-    else 
-        id = await helpers.getTemplateId(templateName)
+    else {
+        id = await getTemplateId(templateName);
+        emailTemplates[templateName] = { id };
+    }
 
-    await helpers.sendEmail(data.to, data.bcc, id, data.params);
+    if (!data['sentTimestamp'])
+        await sendEmail(data.to, data.bcc, data.replyTo, id, data.params);
 
-    return snapshot.ref.update({ sent: true });
+    return snapshot.ref.update({ sentTimestamp: Math.round((new Date()).getTime() / 1000) });
```

Comment:
You're updating this property even if the email was not sent this time. So that's incorrect.
And isn't there any other way of getting the timestamp? Like [this](https://firebase.google.com/docs/reference/android/com/google/firebase/Timestamp)

Reply:
I found this `admin.database.ServerValue.TIMESTAMP`
It's just a placeholder value for auto-populating the current timestamp. It doesn't contain the actual timestamp. The database will replace this placeholder when it executes the command.

Reply:
Good.